### PR TITLE
netty: add return value errorprone annotations

### PR DESCRIPTION
**Pull request by carl-mastrangelo**
Related to #2691.  This requires callers of the netty code to check their return values.  Exceptions are for the Channel builder  and the Server builder, since it's usually okay to not use the result of a setter.  

I eventually would like to add this to more parts of our code, but I'm starting with a small part first.

### CODE REVIEW

**Comment by carl-mastrangelo** [comment]
cc: @ejona86

File: netty/src/test/java/io/grpc/netty/NettyHandlerTestBase.java [hunk 1]
```diff
@@ -221,6 +222,7 @@ protected final Http2Connection connection() {
     return handler().connection();
   }
 
+  @CanIgnoreReturnValue
```
Comment (zhangkun83) [inline netty/src/test/java/io/grpc/netty/NettyHandlerTestBase.java]: Why is this needed?

File: netty/src/main/java/io/grpc/netty/GrpcSslContexts.java [hunk 2]
```diff
@@ -133,6 +134,7 @@ public static SslContextBuilder forServer(
    * Set ciphers and APN appropriate for gRPC. Precisely what is set is permitted to change, so if
    * an application requires particular settings it should override the options set here.
    */
+  @CanIgnoreReturnValue
```
Comment (zhangkun83) [inline netty/src/main/java/io/grpc/netty/GrpcSslContexts.java]: Add `@CheckReturnValue` to the class?

File: netty/src/test/java/io/grpc/netty/NettyHandlerTestBase.java (hunk 1 above)
Comment (carl-mastrangelo) [inline netty/src/test/java/io/grpc/netty/NettyHandlerTestBase.java]: Same reason as in `WriteQueue`, not all callers consume the future.  Netty's API usually returns a future that is passed in.  It looks like this code is trying to be like that.

File: netty/src/main/java/io/grpc/netty/GrpcSslContexts.java (hunk 2 above)
Comment (carl-mastrangelo) [inline netty/src/main/java/io/grpc/netty/GrpcSslContexts.java]: Because it is included in the package, it applies to all classes.  See package-info.java
